lighthouse/helpers/plates.py

```diff
@@ -470,21 +495,174 @@ def create_cherrypicked_post_body(barcode: str, samples: List[Dict[str, Any]]) -
         content = {}
 
         if "control" in sample:
+            content["supplier_name"] = sample["supplier_name"]
             content["control"] = sample["control"]
             content["control_type"] = sample["control_type"]
+            content["uuid"] = sample["uuid"]
         else:
             content["name"] = sample["name"]
             content["phenotype"] = sample["phenotype"]
             content["supplier_name"] = sample["supplier_name"]
             content["sample_description"] = sample["sample_description"]
+            content["uuid"] = sample["uuid"]
 
         wells_content[sample["coordinate"]] = {"content": content}
 
+    subjects = []
+    subjects.append(robot_subject(robot_serial_number))
+    subjects.extend(source_plate_subjects(plate_id_mappings))
+    subjects.extend(sample_subjects(samples))
+
+    events = [
+        {
+            "event": {
+                "user_identifier": user_id,
+                "event_type": PLATE_EVENT_DESTINATION_CREATED,
+                "subjects": subjects,
+                "metadata": {},
+                "lims": app.config["RMQ_LIMS_ID"],
+            }
+        }
+    ]
+
     body = {
         "barcode": barcode,
         "purpose_uuid": app.config["SS_UUID_PLATE_PURPOSE_CHERRYPICKED"],
         "study_uuid": app.config["SS_UUID_STUDY_CHERRYPICKED"],
         "wells": wells_content,
+        "events": events,
     }
 
     return {"data": {"type": "plates", "attributes": body}}
+
+
+def sample_subjects(samples):
+    subjects = []
+    for sample in samples:
+        if "control" in sample:
+            subject = {
+                "role_type": "control",
+                "subject_type": "sample",
+                "friendly_name": control_friendly_name(sample),
+                "uuid": sample["uuid"],
+            }
+        else:
+            subject = {
```

Comment:
Could use the method exposed in `helpers/plate_events` which generates a similar dict. Although looks like the sample is in the correct format for SS, whereas the referred method uses the mongo-format sample

Reply:
To be done in future refactoring on a different branch